public/app/plugins/panel/heatmap-new/HeatmapPanel.tsx

```diff
@@ -35,17 +53,37 @@ export const HeatmapPanel: React.FC<HeatmapPanelProps> = ({
   replaceVariables,
 }) => {
   const theme = useTheme2();
-  const styles = useStyles2(getStyles);
+  // const styles = useStyles2(getStyles);
 
   // ugh
   let timeRangeRef = useRef<TimeRange>(timeRange);
   timeRangeRef.current = timeRange;
 
-  const info = useMemo(() => prepareHeatmapData(data.series, options, theme), [data, options, theme]);
-
+  const info = useMemo(
+    () => prepareHeatmapData(findDataFramesInPanelData(data), options, theme),
+    [data, options, theme]
+  );
+  const exemplars: HeatmapData | undefined = useMemo((): HeatmapData | undefined => {
+    const exemplarsFrame: DataFrame | undefined = findExemplarFrameInPanelData(data);
```

Comment:
can exemplars be part of `prepareHeatmapData`?  

Nicer to move that logic out of the main panel logic